Sources/Commands/SwiftPackageTool.swift

```diff
@@ -198,23 +213,93 @@ extension SwiftPackageTool {
         var packageName: String?
         
         func run(_ swiftTool: SwiftTool) throws {
-            guard let cwd = localFileSystem.currentWorkingDirectory else {
-                throw InternalError("Could not find the current working directory")
+            guard let configPath = try swiftTool.getConfigPath() else {
+                throw InternalError("Could not find config path")
             }
 
-            let packageName = self.packageName ?? cwd.basename
-            let initPackage = try InitPackage(
-                name: packageName, destinationPath: cwd, packageType: initMode)
-            initPackage.progressReporter = { message in
-                print(message)
+            try makePackage(
+                fileSystem: localFileSystem,
+                configPath: configPath,
+                packageName: packageName,
+                mode: .initialize,
+                packageType: packageType,
+                packageTemplate: template)
+        }
+    }
+
+    struct Create: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Create a new package")
+
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+        
+        @Option(name: .customLong("type"))
+        var packageType: InitPackage.PackageType?
+        
+        @Option(help: "Create Package from template")
+        var template: String?
+        
+        @Argument(help: "Package name")
+        var packageName: String
+
+        func run(_ swiftTool: SwiftTool) throws {
+            guard let configPath = try swiftTool.getConfigPath() else {
+                throw InternalError("Could not find config path")
+            }
+            
+            try makePackage(
+                fileSystem: localFileSystem,
+                configPath: configPath,
+                packageName: packageName,
+                mode: .create,
+                packageType: packageType,
+                packageTemplate: template)
+        }
+    }
+    
+    struct AddTemplate: SwiftCommand {
+        static let configuration = CommandConfiguration(
+            abstract: "Donwloads/moves a template into the default template directory")
+        
+        @OptionGroup(_hiddenFromHelp: true)
+        var swiftOptions: SwiftToolOptions
+        
+        @Argument(help: "URL to template (path to template works too)")
+        var url: String
+        
+        func run(_ swiftTool: SwiftTool) throws {
+            guard let configPath = try swiftTool.getConfigPath() else {
+                throw InternalError("Could not find config path")
+            }
+            
+            guard let cwd = localFileSystem.currentWorkingDirectory else {
+                throw InternalError("Could not get the current working directory")
+            }
+            
+            if localFileSystem.exists(cwd.appending(RelativePath(url))) {
```

Comment:
when passing in a path it should be an absolute path, not relative to the working directory.